sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp

```diff
@@ -191,6 +201,69 @@ joint_matrix_mad(Group sg, joint_matrix<T1, M, K, LayoutA, Group> &mA,
                       PI_INVALID_DEVICE);
 #endif // __SYCL_DEVICE_ONLY__
 }
+
+template <typename T, size_t NumRows, size_t NumCols,
+          matrix_layout Layout = matrix_layout::row_major,
+          typename Group = sycl::sub_group>
+class wi_elem {
+  joint_matrix<T, NumRows, NumCols, Layout, Group> &M;
+  std::size_t idx;
+
+public:
+  wi_elem(joint_matrix<T, NumRows, NumCols, Layout, Group> &Mat, std::size_t i)
+      : M(Mat), idx(i) {}
+  operator T() {
+#ifdef __SYCL_DEVICE_ONLY__
+    return __spirv_JointMatrixGetSliceElem(M.spvm, idx);
+#else
+    throw runtime_error("joint matrix is not supported on host device.",
+                        PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+  }
+  wi_elem &operator=(const T &rhs) {
+#ifdef __SYCL_DEVICE_ONLY__
+    M.spvm = __spirv_JointMatrixSetSliceElem(M.spvm, idx, rhs);
```

Comment:
Note that these will be converted to 
SPIRVVectorInsertDynamic
and 
SPIRVVectorextractDynamic
in SPIRV.
and to:
int_experimental_matrix_wi_slice_extractelement
int_experimental_matrix_wi_slice_inserttelement
in LLVM IR
so we need to change the names you are using to insert/extract instead of set/get. Something like:
__spirv_JointMatrix_Extract
__spirv_JointMatrix_Insert
is sufficient to match the SPIRV ones.
What do you think @AlexeySotkin?

Reply:
We should change it to 
```suggestion
    M.spvm = __spirv_VectorInsertDynamic(M.spvm, rhs, idx);
```